jdbc-yugabytedb/src/main/java/com/yugabyte/ysql/ClusterAwareLoadBalancer.java

```diff
@@ -102,16 +118,35 @@ public boolean needsRefresh() {
       hostPortMap_public.put(public_host, port);
       currentPrivateIps.add(host);
```

Comment:
To be on the safer side, can we wrap the following code with a `null` check for variables `host`, `public_host`, and `port`. 
```
      hostPortMap.put(host, port);
      hostPortMap_public.put(public_host, port);
      currentPrivateIps.add(host);
      currentPublicIps.add(public_host);
```
similar to the ones you have here 

```
(host != null
          && !host.isEmpty())
```

Reply:
Actually public_host may be empty but not null according to how the server returns the result. Rest anything cannot be null or empty if the resultset.next is returning true. So let me change it to just do a null check for the public_host.
Please note: One subsequent checkin is going to come where we will change the names of the columns slightly.